packages/react/src/wallet/useCanClaimHandle.ts

```diff
@@ -15,6 +17,14 @@ export type { ClaimableProfilesResult, ReservedClaimable } from '@lens-protocol/
  * @group Hooks
  */
 export function useCanClaimHandle(): ReadResult<ClaimableProfilesResult> {
+  const { data: session } = useSession();
+
+  invariant(session?.authenticated, 'You must be authenticated.');
+  invariant(
+    session.type === SessionType.JustWallet,
+    'You must be authenticated with a WalletOnlySession',
+  );
```

Comment:
Was it failing otherwise?

Reply:
Asking cause this throws in the render path, which at times could be annoying. It can stay if it needs to be.

Reply:
if you have a wrong accessToken then the API returns Unauthorised error, so yes, I think it's better to fail early on the client

Reply:
ok